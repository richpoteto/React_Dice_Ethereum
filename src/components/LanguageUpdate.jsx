import React from 'react';
import { arrayOf, func, string } from 'prop-types';
import { IntlContext } from '../contexts/IntlContext';
import { locales } from '../utils/locales';

const DropdownItem = ({ text, onClick }) => (
  <button className="dropdown-item" type="button" onClick={() => onClick(text)}>{text}</button>
);
DropdownItem.propTypes = {
  text: string.isRequired,
  onClick: func.isRequired,
};

const DropdownMenu = ({ items, onClick }) => (
  <div className="dropdown-menu" aria-labelledby="navbarDropdown">
    {items.map(item => <DropdownItem text={item} onClick={onClick} key={item} />)}
  </div>
);
DropdownMenu.propTypes = {
  items: arrayOf(string).isRequired,
  onClick: func.isRequired,
};

const LanguageUpdate = () => {
  const [locale, setLocale] = React.useContext(IntlContext);

  return (
    <li className="nav-item dropdown">
      <button
        className="nav-link btn button-link dropdown-toggle"
        id="navbarDropdown"
        data-toggle="dropdown"
        aria-haspopup="true"
        aria-expanded="false"
        type="button"
      >
        <i className="fas fa-flag" />
        &nbsp;
        {locale.toUpperCase()}
      </button>
      <DropdownMenu
        items={locales}
        onClick={newLocale => setLocale(newLocale)}
      />
    </li>
  );
};

export default LanguageUpdate;
