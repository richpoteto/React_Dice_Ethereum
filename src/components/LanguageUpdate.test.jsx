import React from 'react';
import renderer from 'react-test-renderer';
import IntlContextProvider from '../contexts/IntlContext';
import LanguageUpdate from './LanguageUpdate';

it('renders correctly', () => {
  const tree = renderer.create(
    <IntlContextProvider>
      <LanguageUpdate />
    </IntlContextProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
