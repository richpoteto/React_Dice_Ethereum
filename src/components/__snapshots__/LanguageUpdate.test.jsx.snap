// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="nav-item dropdown"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="nav-link btn button-link dropdown-toggle"
    data-toggle="dropdown"
    id="navbarDropdown"
    type="button"
  >
    <i
      className="fas fa-flag"
    />
    Â 
    EN
  </button>
  <div
    aria-labelledby="navbarDropdown"
    className="dropdown-menu"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      type="button"
    >
      en
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      type="button"
    >
      cn
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      type="button"
    >
      es
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      type="button"
    >
      fr
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      type="button"
    >
      ru
    </button>
  </div>
</li>
`;
